neutronclient/client.py

```diff
@@ -239,6 +248,8 @@ def authenticate(self):
             self._authenticate_keystone()
         elif self.auth_strategy == 'noauth':
             self._authenticate_noauth()
+        elif self.auth_plugin:
+            self._plugin_auth(self.auth_url)
```

Comment:
It doesn't look like it matters to me. Auth strategy is set by the person using the client, so the others won't be considered if there's an auth plugin being loaded